webapp/app.py

```diff
@@ -30,10 +35,34 @@ def index():
 
 
 # Career departments
-@app.route("/careers/<department>")
+@app.route("/careers/<department>", methods=["GET", "POST"])
 def department_group(department):
     vacancies = get_vacancies(department)
 
+    if flask.request.method == "POST":
```

Comment:
It seems rather over-complicated to submit people back to `request.path`, and then have this fork of this view function which processes data completely differently and then loads a completely different template.

If I were you I would instead create a single `@app.route("/careers/thank-you", methods=["POST"])`, and put all the submission functionality in there.

Reply:
Oh wait the template is the same, so you get returned to the list of careers for that department. Interesting... Maybe this is the best structure?

Reply:
@nottrobin in this case I will leave it as it is, unless you foresee any issues in the future.